TestCases/compliance-level-3/0012-list-functions/0012-list-functions.dmn

```diff
@@ -1,19 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <definitions exporter="DMN Modeler; Method and Style trisofix.xslt" exporterVersion="5.0.33.1; 1.0" namespace="http://www.trisotech.com/definitions/_c0858816-af7b-40a1-8aa7-6e11b8761215" name="listFunctions" triso:logoChoice="Default" id="_c0858816-af7b-40a1-8aa7-6e11b8761215" xmlns="http://www.omg.org/spec/DMN/20180521/MODEL/" xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/" xmlns:dmndi="http://www.omg.org/spec/DMN/20180521/DMNDI/" xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" xmlns:triso="http://www.trisotech.com/2015/triso/modeling" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
-    <itemDefinition isCollection="true" name="tStringList" id="tStringList">
-        <typeRef>string</typeRef>
-    </itemDefinition>
-    <itemDefinition isCollection="true" name="tNumList" id="tNumList">
-        <typeRef>number</typeRef>
-    </itemDefinition>
-    <itemDefinition isCollection="true" name="tListOfLists" id="tListOfLists">
-        <typeRef>tStringList</typeRef>
-    </itemDefinition>
     <inputData name="list1" id="_86f5e710-a139-4cd5-8ac4-90b4671a75b8">
-        <variable typeRef="tStringList" name="list1"/>
```

Comment:
> 0012-list-function: relaxed the typeRefs here. Some tests has incorrect types, and as this test is not testing types, and FEEL does not support lists must varied types nicely, I have removed the typed lists with simply 'list'

I have spent some time to review (manually) CL3 `0012-list-function` but to me all is good with concerning types to me, hence I don't understand why modifications to this test case are required.
If you still believe changes to `0012-list-function` test-case are required, can you kindly describe in a bit more details commenting your proposed changes, please?

Reply:
Hi @tarilabs, some more info over here: https://github.com/dmn-tck/tck/issues/205

The types, I believe, may have been okay in some interpretations of the v1.1 typing system but I think are invalid in v1.2 as the type `[a]` is not the same as type `a`.  If I recall, some results in those tests produced a list, but of mixed types (some string some string array) so therefore, the actual runtime-type type of the list is 'any'.  A list of any is not type compatible with a list of string.  

I think a 1.1 impl would have treated a single string and a list of string as the same type hence the tests were okay for 1.1 - but not for 1.2.

Does that help?

Reply:
I will need to double check that, but then assuming the above, I would suggest the best action is not to elide types from the DMN altogether, but rewrite only those Decisions in the test case which actually exhibit the issue, e.g.: appendListItem, flatten1 (this latter as it depends on appendListItem).

Reply:
Thanks @tarilabs.  My take was as the test assertions here are for the outcome of string functions - not type assertions - then the types have no real role to play here.  If there are cases that do need to be captured regarding type assertions then they should be covered explicitly elsewhere.

To that end, the same assertions are covered (hopefully!) in the `list_*` tests in `0070-feel-instance-of`.  And the coercion to null for mismatching types in `0082-feel-coercion` .